src/staking-v3/components/vote/enter-amount/ChooseAmountsPanel.vue

```diff
@@ -0,0 +1,260 @@
+          </div>
+        </div>
+      </div>
+      <div class="container-2-columns staking-balances">
+        <div>{{ $t('stakingV3.voting.totalStakingAmount') }}</div>
+        <token-balance-native :balance="stakeInfo.totalStakedAmount.toString()" />
+      </div>
+    </div>
+    <error-panel :error-message="stakeInfo.errorMessage" :error-ref-url="stakeInfo.errorRefUrl" />
+    <div class="button-container">
+      <go-back-button v-if="onGoBack" @click="onGoBack && onGoBack()">{{
+        $t('stakingV3.voting.backToCategory')
+      }}</go-back-button>
+      <astar-button :disabled="!canSubmit()" class="submit-button" @click="onAmountsEntered">{{
+        $t('next')
+      }}</astar-button>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, PropType } from 'vue';
+import { DappVote } from '../../../logic';
+import DappIcon from '../DappIcon.vue';
+import Amount from './Amount.vue';
+import TokenBalanceNative from 'src/components/common/TokenBalanceNative.vue';
+import ErrorPanel from '../../ErrorPanel.vue';
+import GoBackButton from '../GoBackButton.vue';
+
+// export type ChoseAmountPanelProps = {
+//   dapps: DappVote[];
+//   totalStakedAmount: bigint;
+//   remainingLockedTokens: bigint;
+//   canSubmit: boolean;
+//   onAmountChanged?: (dapp: DappVote, amount: number) => void;
+//   onContinue: () => void;
+// };
+
+export type StakeInfo = {
+  totalStakedAmount: bigint;
+  remainingLockedTokens: bigint;
+  availableAfterStaking: bigint;
+  errorMessage: string;
+  errorRefUrl: string;
+};
+
+export default defineComponent({
+  components: {
+    DappIcon,
+    Amount,
+    TokenBalanceNative,
+    ErrorPanel,
+    GoBackButton,
+  },
+  props: {
+    dapps: {
+      type: Array as PropType<DappVote[]>,
+      required: true,
+    },
+    onAmountChanged: {
+      type: Function as PropType<(dapp: DappVote, amount: number) => void>,
+      required: false,
+      default: undefined,
+    },
+    stakeInfo: {
+      type: Object as PropType<StakeInfo>,
+      required: true,
+    },
+    canSubmit: {
+      type: Function as PropType<() => boolean>,
+      required: true,
+    },
+    onAmountsEntered: {
+      type: Function as PropType<() => void>,
+      required: true,
+    },
+    onRemoveDapp: {
+      type: Function as PropType<(dapp: DappVote) => void>,
+      required: false,
+      default: undefined,
+    },
+    onGoBack: {
+      type: Function as PropType<() => void>,
+      required: false,
+      default: undefined,
+    },
+  },
+  setup(props) {
+    const handleAmountChanged = (dapp: DappVote, amount: number) => {
+      if (props.onAmountChanged) {
+        props.onAmountChanged(dapp, amount);
+      }
+    };
+
+    const handleRemoveDapp = (dapp: DappVote) => {
+      if (props.onRemoveDapp) {
+        props.onRemoveDapp(dapp);
+      }
+    };
+
+    return { handleAmountChanged, handleRemoveDapp };
+  },
+});
+</script>
+
+<style lang="scss" scoped>
```

Comment:
I think the number of lines in this file is a bit too many; perhaps it's better to separate the styling code from this file (and some other files too).

Reply:
Thanks for comments. Fixed in https://github.com/AstarNetwork/astar-apps/pull/1291/commits/200eee81e648cad59d98ff69aae9658a649833c6